neutron-physics/model.py

```diff
@@ -0,0 +1,330 @@
+        self.name = name
+
+    @property
+    def energy_mev(self):
+        return self.energy*1.e-6
+
+    def zaid(self, nuclide, suffix):
+        Z, A, m = openmc.data.zam(nuclide)
+
+        # Correct the ground state and first excited state of Am242, which are
+        # the reverse of the convention
+        if A == 242 and m == 0:
+            m = 1
+        elif A == 242 and m == 1:
+            m = 0
+
+        if m > 0:
+            A += 300 + 100*m
+
+        if re.match('(71[0-6]nc)', suffix):
+            suffix = f'8{suffix[2]}c'
+
+        return f'{1000*Z + A}.{suffix}'
+
+    def szax(self, nuclide, suffix):
+        Z, A, m = openmc.data.zam(nuclide)
+
+        # Correct the ground state and first excited state of Am242, which are
+        # the reverse of the convention
+        if A == 242 and m == 0:
+            m = 1
+        elif A == 242 and m == 1:
+            m = 0
+
+        if re.match('(7[0-4]c)|(8[0-6]c)', suffix):
+            suffix = f'71{suffix[1]}nc'
+
+        return f'{1000000*m + 1000*Z + A}.{suffix}'
+
+    def _create_library(self):
+        """Convert the ENDF ACE data from the MCNP distribution into an HDF5
+        library that can be used by OpenMC.
+
+        """
+        if re.match('7[0-4]c', self.suffix):
+            # Get the table from the ENDF/B-VII.0 neutron ACE files
+            zaid = self.zaid(self.nuclide, self.suffix)
+            for path in Path(self.library).glob('endf70[a-k]'):
+                try:
+                    table = openmc.data.ace.get_table(path, zaid)
+                except ValueError:
+                    pass
+        else:
+            # Get the table from the ENDF/B-VII.1 neutron ACE files
+            Z, A, m = openmc.data.zam(self.nuclide)
+            element = openmc.data.ATOMIC_SYMBOL[Z]
+            szax = self.szax(self.nuclide, self.suffix)
+            path = Path(self.library) / f'endf71x/{element}/{szax}'
+            table = openmc.data.ace.get_table(path, szax)
+
+        # Convert cross section data
+        data = openmc.data.IncidentNeutron.from_ace(table, 'mcnp')
+
+        # Export HDF5 file
+        os.makedirs('openmc', exist_ok=True)
+        h5_file = Path(f'openmc/{data.name}.h5')
+        data.export_to_hdf5(h5_file, 'w')
+
+        # Register with library and write cross_sections.xml
+        data_lib = openmc.data.DataLibrary()
+        data_lib.register_file(h5_file)
+        data_lib.export_to_xml(Path('openmc/cross_sections.xml'))
+
+    def _make_openmc_input(self):
+        """Generate the OpenMC input XML
+
+        """
+        # Directory from which openmc is run
+        os.makedirs('openmc', exist_ok=True)
+        
+        # Define material
+        mat = openmc.Material()
+        mat.add_nuclide(self.nuclide, 1.0)
+        mat.set_density('g/cm3', self.density)
+        materials = openmc.Materials([mat])
+        if self.library is not None:
+            xs_path = Path('openmc/cross_sections.xml').resolve()
```

Comment:
```Python
(Path('openmc') / 'cross_sections.xml').resolve()
```

Reply:
I think you get the idea (use `/` operator as opposed to embedding it in a string)